metaci/plan/migrations/0022_migrate_to_groups.py

```diff
@@ -64,7 +62,15 @@ def create_groups_and_migrate_users(apps, schema_editor):
 
 def grant_anonymous_perms(apps, schema_editor):
     PlanRepository = apps.get_model('plan.PlanRepository')
+    Group = apps.get_model('auth.Group')
     User = apps.get_model('users.User')
+    Permission = apps.get_model('auth.Permission')
+    ContentType = apps.get_model('contenttypes', 'ContentType')
+    GroupObjectPermission = apps.get_model('guardian.GroupObjectPermission')
+
+    public_group = Group.objects.create(name='Public')
+    view_builds = Permission.objects.get(codename='view_builds')
+
     try:
         anon = get_anonymous_user()
     except Exception as e:
```

Comment:
can just catch User.DoesNotExist ?

Reply:
Yeah, migrations are kinda the twilight zone when it comes to models.  The error that's thrown is actually metaci.users.models.DoesNotExist but if you try to build an except using that path you get a ModuleError since that doesn't exist.  There's some real "magic" in how migrations work which was a PITA when figuring this stuff out, but I've tested this against a clone of the mrbelvedereci-staging database and against a clean database.

Reply:
I *think* `ObjectDoesNotExist` should catch it (And any DoesNotExist-ish errors), but yea, if this is what it takes.

Reply:
It does, pushed new commit using ObjectDoesNotExist